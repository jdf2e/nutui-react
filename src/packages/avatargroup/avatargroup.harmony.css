.nut-avatar-group {
  background-size: 100% 100%;
  background-repeat: no-repeat;
  background-position: center center;
  display: inline-block;
  position: relative;
  flex: 0 0 auto;
}
.nut-avatar-group .nut-avatar {
  border: 1px solid #fff;
}
.nut-avatar-group .nut-avatar:not(:first-of-type) {
  margin-left: -8px;
}

[dir=rtl] .nut-avatar-group .nut-avatar:not(:first-of-type),
.nut-rtl .nut-avatar-group .nut-avatar:not(:first-of-type) {
  margin-left: 0;
  margin-right: -8px;
}