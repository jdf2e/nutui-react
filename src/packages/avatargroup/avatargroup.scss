.nut-avatar-group {
  background-size: 100% 100%;
  background-repeat: no-repeat;
  background-position: center center;
  display: inline-block;
  position: relative;
  flex: 0 0 auto; // 防止被压缩
  .nut-avatar {
    border: 1px solid #fff;
    &:not(:first-of-type) {
      margin-left: -8px;
    }
  }
}
