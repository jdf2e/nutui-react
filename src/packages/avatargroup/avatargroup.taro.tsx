import React, { FunctionComponent, useRef } from 'react'
import classNames from 'classnames'
import { View } from '@tarojs/components'
import { AvatarContext } from './context'
import { BasicComponent, ComponentDefaults } from '@/utils/typings'

export interface AvatarGroupProps extends BasicComponent {
  maxContent: string
  max: string | number
  maxBackground: string
  maxColor: string
  size: 'large' | 'normal' | 'small'
  shape: 'round' | 'square'
  gap: string
  level: 'left' | 'right'
}
const defaultProps = {
  ...ComponentDefaults,
  maxContent: '',
  max: '',
  maxBackground: '#eee',
  maxColor: '#666',
  gap: '-8',
  level: 'left',
} as AvatarGroupProps

const classPrefix = `nut-avatar-group`
export const AvatarGroup: FunctionComponent<
  Partial<AvatarGroupProps> & React.HTMLAttributes<HTMLDivElement>
> = (props) => {
  const propAvatarGroup = { ...defaultProps, ...props }
  const { className, style, children } = propAvatarGroup

  const avatarGroupRef = useRef(null)
  const cls = classNames(classPrefix, className)

  const parentAvatar = {
    propAvatarGroup: {
      avatarCount: React.Children.count(children) || 0,
      ...propAvatarGroup,
    },
    avatarGroupRef,
  }

  return (
    <AvatarContext.Provider value={parentAvatar}>
      <View className={cls} style={style} ref={avatarGroupRef}>
        {React.Children.map(children, (child, index) => {
          // @ts-ignore
          return child?.type?.displayName === 'NutAvatar'
            ? // @ts-ignore
              React.cloneElement(child, {
                avatarIndex: index + 1,
                className: 'nut-avatar-group-avatar',
              })
            : child
        })}
      </View>
    </AvatarContext.Provider>
  )
}

AvatarGroup.displayName = 'NutAvatarGroup'
