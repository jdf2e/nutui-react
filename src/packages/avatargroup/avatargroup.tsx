import React, { FunctionComponent, useRef } from 'react'
import classNames from 'classnames'
import { AvatarContext } from './context'

import { BasicComponent, ComponentDefaults } from '@/utils/typings'

export interface AvatarGroupProps extends BasicComponent {
  maxContent: string
  max: string | number
  maxBackground: string
  maxColor: string
  size: 'large' | 'normal' | 'small'
  shape: 'round' | 'square'
  gap: string
  level: 'left' | 'right'
}
const defaultProps = {
  ...ComponentDefaults,
  maxContent: '',
  max: '',
  maxBackground: '#eee',
  maxColor: '#666',
  gap: '-8',
  level: 'left',
} as AvatarGroupProps

const classPrefix = `nut-avatar-group`
export const AvatarGroup: FunctionComponent<
  Partial<AvatarGroupProps> & React.HTMLAttributes<HTMLDivElement>
> = (props) => {
  const propAvatarGroup = { ...defaultProps, ...props }
  const { className, style, children } = propAvatarGroup

  const avatarGroupRef = useRef(null)
  const cls = classNames(classPrefix, className)

  const parentAvatar = {
    propAvatarGroup,
    avatarGroupRef,
  }
  return (
    <AvatarContext.Provider value={parentAvatar}>
      <div className={cls} style={style} ref={avatarGroupRef}>
        {children}
      </div>
    </AvatarContext.Provider>
  )
}

AvatarGroup.defaultProps = defaultProps
AvatarGroup.displayName = 'NutAvatarGroup'
