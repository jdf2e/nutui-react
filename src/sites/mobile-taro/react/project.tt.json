{
  "miniprogramRoot": "./",
  "projectname": "first",
  "appid": "testAppId",
  "setting": {
    "es6": false,
    "minified": false
  }
}
