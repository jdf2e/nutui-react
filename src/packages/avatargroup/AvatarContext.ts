import { createContext } from 'react'
export const AvatarContext = createContext({})
