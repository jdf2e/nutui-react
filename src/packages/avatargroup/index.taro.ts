import { AvatarGroup } from './avatargroup.taro'

export default AvatarGroup
