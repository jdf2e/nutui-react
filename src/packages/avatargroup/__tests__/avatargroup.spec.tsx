import * as React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'

import { AvatarGroup } from '../avatargroup'

test('size prop', () => {
  const { container } = render(<AvatarGroup size="small"></AvatarGroup>)
  expect(container).toMatchSnapshot()
})
