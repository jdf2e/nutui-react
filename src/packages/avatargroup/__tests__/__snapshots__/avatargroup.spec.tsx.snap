// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`size prop 1`] = `
<div>
  <div
    class="nut-avatar-group"
  />
</div>
`;
